### ISSUE TITLE
Remote screen (via ssh) crashes BashOnWindows

### ISSUE THREAD

**Issue by WindfallProphet** [Microsoft/BashOnWindows#100]
Steps to Reproduce:
1. ssh into remote machine
2. run screen on remote machine

**Comment by eternelpanic** [comment]
Works for me.

**Comment by hut8** [comment]
Might want to try Cmder and set Bash up as a task:

```
C:\Windows\System32\bash.exe ~
```

I think specifically, the issue you're having isn't with `bash` per se, but rather with your "terminal emulator" which is just `cmd.exe`. `tmux`, at least, works great for me over ssh currently (with minor color issues which are due to my hacks from using previously using gnome-terminal I think).

`cmd.exe` has come a very long way, especially with Windows 10, but it has a lot of issues.

**Comment by MrObvious** [comment]
I'm getting this too.
Win+R, bash, OK
Maximize Window
ssh user@server
server$ screen -x
(Window closes)

If you don't maximize bash it seems to work.

**Comment by bitcrazed** [comment]
Investigating

**Comment by zadjii-msft** [comment]
So I'm seeing a bunch of related but separate issues here so let me see if I can't work them all out:

* Running `screen` while connected to a remote machine with ssh crashes conhost.exe
  - This looks like it's fixed in recent builds. 
* Ctrl-A-D terminates instead of detaches
* `screen -x` crashes conhost instead of reattaching
  - These two look like a known issue causing the console window to crash with certain ncurses applications. Fortunately, the fix is being tested already!

**Comment by pqnet** [comment]
affects me too. Using another ssh client (i.e., the one distributed with git for windows) inside cmd.exe works. Logs report a crash of conhost.exe:

```
Faulting application name: conhost.exe, version: 10.0.14332.1001, time stamp: 0x571aff9e
Faulting module name: ConhostV2.dll, version: 10.0.14332.1001, time stamp: 0x571afe8d
Exception code: 0xc0000005
Fault offset: 0x000000000000a5c7
Faulting process id: 0x2890
Faulting application start time: 0x01d1a7acf5db68e2
Faulting application path: C:\WINDOWS\system32\conhost.exe
Faulting module path: C:\WINDOWS\SYSTEM32\ConhostV2.dll
Report Id: f0eed5f4-1091-498a-9149-240e5f280ca6
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by xilun** [comment]
Hi,
Is this supposed to be fixed in 14352? I'm having conhost crashes in other conditions on 14352 and would like to know if I need to wait for this fix or report it.
Thanks!

**Comment by xilun** [comment]
Note: I just tested screen on a remote ssh Linux from WSL bash and this bug might not be fixed on 14352, or incompletely. The first time the console crashed, and on two or three subsequent tries it worked.

**Comment by dirk1978** [comment]
Still doesn't seem to be fixed in 14352.1002. For me it manifests as crashing the terminal when closing vim, which I assume does the same tput reset and tput init bug as https://github.com/Microsoft/BashOnWindows/issues/399 and https://github.com/Microsoft/BashOnWindows/issues/400
Any ideas when the fix for this will be in a build? Makes WSL pretty much unusable for me and I have been wanting to dump putty and mremote for ages.

**Comment by benhillis** [comment]
I'm virtually certain this is resolved in build 14388+

Please reopen if you continue to see this issue, thanks!
